Development/pcm_block.yml

```diff
@@ -0,0 +1,64 @@
+name: ConsIndShock
+
+blocks:
+  - name: Consumption
+    symbols:
+      states: [m]
+      controls: [c]
+      parameters: [beta, rho]
+    domain:
+      m: [0, .inf]
+      c: [0, .inf]
+    equations:
+      transitions:
+        - a: m - c
+        - v_0(m): u(c) + beta * v_1(a)
+    definitions:
+      - u(c): c^(1-rho) / (1-rho)

```

Comment:
are rewards special, or just a kind of definition/symbol substitution?

Reply:
In this abstract formulation, rewards are not particularly special. 

Rewards are a component of "Bellman Equation" aggregators, ie reward + disc. continuation. 

```
aggregator: !Bellman
   reward: u
   discount: beta
   continuation: w
```

This is a quick sketch of something I have in mind for Abstract Dynamic Programming. In ADP, there might be different types of aggregators, with different components and different structures. However, in the aggregator-less language description, we can describe any aggregator as transitions of states and values.